src/Request/Init.php

```diff
@@ -129,10 +130,24 @@ class Init
      * @throws ValidationException
      */
     public function __construct(
+        string $service = null,
+        $securityPacket = null,
+        string $secret = null,
+        $requestPacket = null,
+        string $action = null,
+        SignatureFactory $signatureFactory = null
+    ) {
+        if (is_null($service)) {
```

Comment:
@klauste We should probably address codacy one given it makes sense.

Reply:
Also could we write a helper function to check all parameters require are not null and not just service? I worry that reading the constructor signature gives the impression that null is acceptable for all parameters